c2/c2-client-bundle/c2-client-service/src/test/java/org/apache/nifi/c2/client/service/operation/UpdateConfigurationOperationHandlerTest.java

```diff
@@ -0,0 +1,121 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.c2.client.service.operation;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.nifi.c2.client.service.operation.UpdateConfigurationOperationHandler.LOCATION;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import org.apache.nifi.c2.client.api.C2Client;
+import org.apache.nifi.c2.client.service.FlowIdHolder;
+import org.apache.nifi.c2.protocol.api.C2Operation;
+import org.apache.nifi.c2.protocol.api.C2OperationAck;
+import org.apache.nifi.c2.protocol.api.C2OperationState;
+import org.apache.nifi.c2.protocol.api.OperandType;
+import org.apache.nifi.c2.protocol.api.OperationType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class UpdateConfigurationOperationHandlerTest {
+
+    private static final String FLOW_ID = "flowId";
+    private static final String OPERATION_ID = "operationId";
+    private static final Map<String, String> CORRECT_LOCATION_MAP = Collections.singletonMap(LOCATION, "/path/for/the/" + FLOW_ID);
+    private static final Map<String, String> INCORRECT_LOCATION_MAP = Collections.singletonMap(LOCATION, "incorrect/location");
+
+    @Mock
+    private C2Client client;
+
+    @Mock
+    private FlowIdHolder flowIdHolder;
+
+    @Test
+    void testUpdateConfigurationOperationHandlerCreateSuccess() {
+        UpdateConfigurationOperationHandler handler = new UpdateConfigurationOperationHandler(null, null, null);
+
+        assertEquals(OperationType.UPDATE, handler.getOperationType());
+        assertEquals(OperandType.CONFIGURATION, handler.getOperandType());
+    }
+
+    @Test
+    void testHandleThrowsExceptionForIncorrectArg() {
+        UpdateConfigurationOperationHandler handler = new UpdateConfigurationOperationHandler(null, null, null);
+        C2Operation operation = new C2Operation();
+        operation.setArgs(INCORRECT_LOCATION_MAP);
+
+        IllegalStateException exception = assertThrows(IllegalStateException.class, () -> handler.handle(operation));
+
+        assertEquals("Could not get flow id from the provided URL", exception.getMessage());
```

Comment:
Recommend adjusting this check to avoid testing for the exact exception message.

Reply:
As we are already matching the thrown exception I removed this assertion it made it more fragile with not much added value.